Framework/Kernel/src/MultiFileNameParser.cpp

```diff
@@ -461,96 +455,79 @@ namespace // anonymous
  *
  * @param parsedRuns :: the vector of vectors of runs parsed so far.
  * @param token      :: the token to parse.
- *
- * @returns the newly parsed runs appended to the previously parsed runs.
- * @throws std::runtime_error if
  */
-std::vector<std::vector<unsigned int>> &
-parseToken(std::vector<std::vector<unsigned int>> &parsedRuns,
-           const std::string &token) {
-  // Tokenise further, on plus, minus or colon.
+void parseToken(std::vector<std::vector<unsigned int>> &parsedRuns,
```

Comment:
Line 451 isn't correct any more - you can't use this method with `std::accumulate`. What was the reason behind making the function non `accumulatable`-able?

Reply:
To me the way `std::accumulate` was (ab)used with this function was confusing. What is needed is just a simple loop over the tokens, see lines 132-134. I just forgot to update the doxygen so its good that you noticed.